micropip/package_index.py

```diff
@@ -234,7 +238,13 @@ def _select_parser(content_type: str, pkgname: str) -> Callable[[str], ProjectIn
         ):
             return partial(ProjectInfo.from_simple_html_api, pkgname=pkgname)
         case _:
-            raise ValueError(f"Unsupported content type: {content_type}")
+            raise UnsupportedContentTypeError(
+                f"Unsupported content type: {content_type}"
+            )
+
+
+class NoValidIndexForPackageError(Exception):
```

Comment:
Maybe `PackageNotFoundError` would be better? `NoValidIndex` is a bit confusing IMO.

Reply:
That would be a synonym of PackageNotFoundError from importlib.metadata which is used un the same file... I've used a slightly longer name to differentiate.